sway-core/src/lib.rs

```diff
@@ -679,8 +679,11 @@ fn parse_root_from_pairs(
         }
     }
 
-    let fuel_ast = fuel_ast_opt.unwrap();
-    ok(fuel_ast, warnings, errors)
+    CompileResult {
+        value: fuel_ast_opt,
+        warnings,
+        errors,
+    }
```

Comment:
`ok()` constructs a compile result -- it is preferable to use that.
```suggestion

    ok(fuel_ast_op, warnings, errors)
```

Reply:
Ah, I see. Could you add a method in `error.rs`, perhaps `pub fn new()`, that takes an option, warnings, and errors, and makes a compile result? That way we can use it in the future as well.